test/Prompt/DontWrap.C

```diff
@@ -172,7 +172,7 @@ const Test2::Nested::Value::type & Test2::Nested::reference() const {
 Test2::Nested Nest2(80);
 // CHECK: Nested::Nested(160)
 Nest2.simpleAdd(3)
-// CHECK: (Test2::Nested::Value::type) 83
+// CHECK: (int) 83
```

Comment:
Testing for a builtin type is easy and is already being done (in the PR).
Whats complicated is delineating between typedefs that are better desugared and those better left alone.

From this thread I have another commit that does this:
\[cling\]$ std::vector\<int\> b(1)
(std::vector\<int\> &) { 0 }
\[cling\]$ b\[0\]
(int) 0
\[cling\]$ b.begin()
(std::vector\<int, std::allocator\<int\> \>::iterator) @0x7fad5d062910
\[cling\]$ 
\[cling\]$ class Test {};
\[cling\]$ typedef Test Test2;
\[cling\]$ Test2 t
(Test2 &) @0x105f42ffc
\[cling\]$ 
\[cling\]$ std::vector\<Test\> tb(1)
(std::vector\<Test\> &) { @0x7fad5d1031b0 }
\[cling\]$ tb\[0\]
(Test &) @0x7fad5d1031b0
\[cling\]$ tb.begin()
(std::vector\<Test, std::allocator\<Test\> \>::iterator) @0x7fad5a8e7440
\[cling\]$ 
\[cling\]$ std::vector\<Test2\> tb2(1)
(std::vector\<Test2\> &) { @0x7fad5a912860 }
\[cling\]$ tb2\[0\]
(Test &) @0x7fad5a912860
\[cling\]$ tb2.begin()
(std::vector\<Test, std::allocator\<Test\> \>::iterator) @0x7fad5aa0ce80
\[cling\]$ 
\[cling\]$ typedef std::vector\<Test2\> TVec;
\[cling\]$ TVec tv
(TVec &) {}

Where current behavior is this:
\[cling\]$ std::vector\<int\> b(1)
(std::vector\<int\> &) @0x106eb3488
\[cling\]$ b\[0\]
(std::__1::__vector_base\<int, std::__1::allocator\<int\> \>::value_type) 0
\[cling\]$ b.begin()
(std::__1::vector\<int, std::__1::allocator\<int\> \>::iterator) @0x7ff61d777ac0
\[cling\]$ 
\[cling\]$ class Test {};
\[cling\]$ typedef Test Test2;
\[cling\]$ Test2 t
(Test2 &) @0x106eb3ffc
\[cling\]$ 
\[cling\]$ std::vector\<Test\> tb(1)
(std::vector\<Test\> &) @0x106eb5688
\[cling\]$ tb\[0\]
(std::__1::__vector_base\<Test, std::__1::allocator\<Test\> \>::value_type &) @0x7ff61f044c70
\[cling\]$ tb.begin()
(std::__1::vector\<Test, std::__1::allocator\<Test\> \>::iterator) @0x7ff61f045e90
\[cling\]$ 
\[cling\]$ std::vector\<Test2\> tb2(1)
(std::vector\<Test2\> &) @0x106eb5ba0
\[cling\]$ tb2\[0\]
(std::__1::__vector_base\<Test, std::__1::allocator\<Test\> \>::value_type &) @0x7ff61ad88b10
\[cling\]$ tb2.begin()
(std::__1::vector\<Test, std::__1::allocator\<Test\> \>::iterator) @0x7ff61f153680
\[cling\]$ 
\[cling\]$ typedef std::vector\<Test2\> TVec;
\[cling\]$ TVec tv
(TVec &) @0x108732fc8